app/models/user.rb

```diff
@@ -24,6 +24,7 @@ class User < ApplicationRecord
   has_many :comments, dependent: :nullify
   has_many :comment_threads_deleted, class_name: 'CommentThread', foreign_key: :deleted_by_id, dependent: :nullify
   has_many :comment_threads_locked, class_name: 'CommentThread', foreign_key: :locked_by_id, dependent: :nullify
+  has_many :filters, dependent: :destroy
   belongs_to :deleted_by, required: false, class_name: 'User'
```

Comment:
On another note, the question is why category_filter_defaults is linked to user, since it is also linked to filter and filter is linked to a user.

Reply:
The problem is the System filters. We could have, for instance

Filter(Positive, belongs to System)
CategoryFilterDefault(Moshi, Meta, Positive)

To be honest, attaching "universal" filters to System might not have been the brightest idea, since it leads to weird edge cases like this.